src/Dimensions/dimunitrange.jl

```diff
@@ -0,0 +1,48 @@
+struct DimUnitRange{T,R<:AbstractUnitRange{T},D<:Dimension} <: AbstractUnitRange{T}
+    range::R
+    dim::D
+end
+
+DimUnitRange{T}(r::DimUnitRange{T}) where {T<:Integer} = r
+function DimUnitRange{T}(r::DimUnitRange) where {T<:Integer}
+    return DimUnitRange(AbstractUnitRange{T}(parent(r)), dims(r))
+end
+
+@inline Base.parent(r::DimUnitRange) = r.range
+
+@inline dims(r::DimUnitRange) = r.dim
+@inline dims(rs::Tuple{DimUnitRange,Vararg{DimUnitRange}}) = map(dims, rs)
+
+# this is necessary to ensure that keyword syntax for DimArray works correctly
+Base.Slice(r::DimUnitRange) = Base.Slice(parent(r))
+
+Base.show(io::IO, r::DimUnitRange) = print(io, DimUnitRange, (r.range, r.dim))
+
+# the below are adapted from OffsetArrays
+# https://github.com/JuliaArrays/OffsetArrays.jl/blob/master/src/axes.jl
+
+for f in [:length, :isempty, :first, :last]
+    @eval @inline Base.$f(r::DimUnitRange) = Base.$f(parent(r))
+end
+@inline Base.axes(r::DimUnitRange) = (r,)
+@inline Base.axes1(r::DimUnitRange) = r
+if VERSION < v"1.8.2"
```

Comment:
Why the version check here? maybe add a comment

Reply:
It was due to it being unnecessary on recent Julia versions and increasing invalidations. I added a comment.